### PR TITLE
TYP: resample

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @jbrockmendel generally lgtm.

File: pandas/core/resample.py [hunk 1]
```diff
@@ -1822,8 +1823,13 @@ def _take_new_index(
 
 
 def _get_timestamp_range_edges(
-    first, last, freq, closed="left", origin="start_day", offset=None
-):
+    first: Timestamp,
+    last: Timestamp,
+    freq: BaseOffset,
+    closed: str = "left",
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/resample.py:1829]: good idea, will do

File: pandas/core/resample.py [hunk 2]
```diff
@@ -1822,8 +1823,13 @@ def _take_new_index(
...
+    origin="start_day",
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/resample.py:1830]: not trivial bc the validation in the caller suggests the docstring may not be accurate.  id prefer to sort that out separately

File: pandas/core/resample.py [hunk 3]
```diff
@@ -1053,8 +1055,7 @@ def __init__(self, obj, *args, **kwargs):
         for attr in self._attributes:
             setattr(self, attr, kwargs.get(attr, getattr(parent, attr)))
 
-        # error: Too many arguments for "__init__" of "object"
-        super().__init__(None)  # type: ignore[call-arg]
+        self.binner = kwargs.get("binner", getattr(parent, "binner"))
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/resample.py:1058]: im sure past-me had a reason, but the jerk didn't document it so i cant remember.  will revert.

File: pandas/core/resample.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/core/resample.py:1835]: sure

**Review by simonjayhawkins — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**